### PR TITLE
Add validation workflow using COGS

### LINKED ISSUE #70: Write a table validation script
We need a validation script for new cohort mappings, like the ones in the `templates/` directory.

This will be a small python script that looks at the contents based on the headers described in #62.

As far as workflow goes:
1. User fills out the Google Sheet
2. Run `cogs fetch && cogs pull` to update the local version (DROID button to kick this off, includes steps 2-5)
3. Run this script over the local version to output a standardized problems table
4. Run `cogs apply` with the standardized problems table
5. Run `cogs push` to update the remote version of the sheet
6. The user reviews the Google Sheet for any problems, fixes them, and repeats the process if necessary

This script should check for things like:
* Parent term exists as a label
* GECKO Category is an actual term from GECKO
* No duplicate labels

Perhaps warnings or info messages on missing GECKO categories? The user may not be able to map all their terms to existing categories, so it should be OK to leave them blank...

Would it make sense to also do automated ID assignment with this script, or is that handled elsewhere?

### PR DESCRIPTION
**Pull request by beckyjackson**
Resolves #70 

This assumes:
1. The cohort has selected a namespace
2. We are on a branch that is named with that namespace (`git checkout -b <namespace>`)
3. There is a `GOOGLE_CREDENTIALS` environment variable set

First we set up the COGS project:
```
make EMAIL=<user-email> init-cogs
```

Right now the email is just passed to `make`, but maybe we want to set that as an env var as well? I don't know what's best.

The last line of the `init-cogs` task is the link to the new Google Sheet. It will have one sheet (the branch name/namespace) with the required headers described in #62.

The user adds their terms & any mappings that they can do, then:
```
make apply-problems
```

The user then looks at the sheet again to see if any problems were found.

The `validate.py` script checks:
* All "Parent Term" values are defined as labels somewhere in the table
* All "GECKO Category" values are valid IHCC browser labels
* No duplicate labels

It also assigns numeric IDs starting after whatever the last-used numeric ID was. It will start at `ns:0000001` if there are no existing IDs, or no existing numeric IDs (some cohorts already have IDs that use text).

This will also work for updating cohorts, as long as we are on a branch with that cohort name. For example, if we want to update GCS:
```
git checkout -b gcs
make EMAIL=myemail@email.com init-cogs
```

And the sheet will contain the existing `templates/gcs.tsv` that can be updated & validated.

Once we are finished editing and validating, the cogs directory & corresponding Google Sheet should be removed:
```
make destroy-cogs
```

### CODE REVIEW

**Comment by beckyjackson** [comment]
Do we want the `validate.py` script to automatically fix things like lowercase definition, leading & trailing whitespace, etc.?

**Comment by jamesaoverton** [comment]
For this purpose:

- yes, trim leading and trailing whitespace automatically
- no, don't lowercase automatically -- I'm not even sure that should be a warning for this

File: Makefile [hunk 1]
```diff
@@ -153,6 +153,45 @@ data/full-cohort-data.json: data/cohort-data.json data/random-data.json
...
+
+init-cogs: .cogs
+
+templates/$(BRANCH).tsv:
+	echo -e "Term ID\tLabel\tParent Term\tDefinition\tGECKO Category\tSuggested Categories\tComment\nID\tLABEL\tC % SPLIT=|\tA definition\n\tis-required;" > $@
+
+# required env var GOOGLE_CREDENTIALS
+.cogs: | templates/$(BRANCH).tsv
+	cogs init -u $(EMAIL) -t $(BRANCH)
+	cogs add $< -r 2
```
Comment (jamesaoverton) [inline Makefile:169]: This doesn't add anything: there's no value for `$<` because stuff after the `|` does not count.

File: Makefile [hunk 2]
```diff
@@ -153,6 +153,45 @@ data/full-cohort-data.json: data/cohort-data.json data/random-data.json
...
+	cogs open
+
+destroy-cogs: | .cogs
+	cogs delete -f
+
+
+### Validation
+
+build/gecko_labels.tsv: build/gecko.owl src/queries/get_labels.rq | build/robot.jar
+	$(ROBOT) query --input $< --query $(word 2,$^) $@
```
Comment (jamesaoverton) [inline Makefile:180]: `make: *** No rule to make target 'src/queries/get_labels.rq', needed by 'build/gecko_labels.tsv'.  Stop.  `

File: Makefile (hunk 1 above)
Comment (beckyjackson) [reply Makefile]: Fixed

File: Makefile (hunk 2 above)
Comment (beckyjackson) [reply Makefile]: Changed to `robot export`

**Comment by jamesaoverton** [comment]
I want to do more of this work in a Python script that's friendlier to DROID, but this is a good starting point so I'm going to merge it.

**Pull request merged**